dpnp/dpnp_iface_histograms.py

```diff
@@ -306,6 +312,61 @@ def digitize(x, bins, right=False):
     return bins.size - dpnp.searchsorted(bins[::-1], x, side=side)
 
 
+def _find_supported_dtype(dt, supported):
+    if dt in supported:
+        return dt
+
+    for t in supported:
+        if dpnp.can_cast(dt, t):
+            return t
+
+    return None
+
+
+def _result_type_for_device(dtype1, dtype2, device):
+    rt = dpnp.result_type(dtype1, dtype2)
+    return map_dtype_to_device(rt, device)
+
+
+def _align_dtypes(a_dtype, bins_dtype, ntype, device):
+    has_fp64 = device.has_aspect_fp64
+    a_bin_dtype = _result_type_for_device(a_dtype, bins_dtype, device)
+
+    supported_types = (dpnp.float32, dpnp.int64, numpy.uint64, dpnp.complex64)
+    if has_fp64:
+        supported_types += (dpnp.float64, dpnp.complex128)
+
+    float_types = [dpnp.float32, dpnp.float64]
+    complex_types = [dpnp.complex64, dpnp.complex128]
+
+    a_bin_dtype = _find_supported_dtype(a_bin_dtype, supported_types)
+    hist_dtype = _find_supported_dtype(ntype, supported_types)
+    if hist_dtype == numpy.uint64:
+        hist_dtype = dpnp.int64
+
+    if (a_bin_dtype in float_types and hist_dtype in float_types) or (
```

Comment:
Should it be `else if`?
```suggestion
    elif (a_bin_dtype in float_types and hist_dtype in float_types) or (
```

Reply:
No. `a_bin_dtype` is not depends on `hist_dtype`